### ISSUE TITLE
Types of property 'css' are incompatible. Type 'InterpolationWithTheme<any>' is not assignable to type 'Interpolation<Theme>'

### ISSUE THREAD

**Issue by Vadorequest** [emotion-js/emotion#2320]
**Current behavior:**

When using `css` prop with Next.js, it sometimes yields a warning about `Types of property 'css' are incompatible.`. (not always)

I'm confused as to why. For now, [I'm `// @ts-ignore` all issues manually](https://github.com/UnlyEd/next-right-now/pull/304/commits/547ef9175ac30045309c4919ba602aadfedd4693).

It happens on both `svg` elements and `css` properties.

```
Error:(51, 6) TS2322: Type '{ accept?: string; acceptCharset?: string; action?: string; allowFullScreen?: boolean; allowTransparency?: boolean; alt?: string; as?: string; async?: boolean; autoComplete?: string; autoFocus?: boolean; ... 356 more ...; flipModeOptions: { ...; }; }' is not assignable to type '{ css?: Interpolation<Theme>; }'.
  Types of property 'css' are incompatible.
    Type 'InterpolationWithTheme<any>' is not assignable to type 'Interpolation<Theme>'.
```

![image](https://user-images.githubusercontent.com/3807458/113160450-cc1c5000-923d-11eb-8b99-5baa50412eba.png)

![image](https://user-images.githubusercontent.com/3807458/113162633-c162ba80-923f-11eb-900d-6bf8c4b81e1d.png)

**To reproduce:**

1. `git clone https://github.com/UnlyEd/next-right-now.git at-update-deps-03-31`
1. `cd at-update-deps-03-31 && git checkout at-update-deps-03-31`
1. `cp .env.local.example .env.local` - Uses the default ENV variables when running locally
1. `yarn`
1. `yarn start`
1. Open `http://localhost:8888`

PR: https://github.com/UnlyEd/next-right-now/pull/304

**Expected behavior:**

It should't issue an error.

**Environment information:**

    "@emotion/react": "11.1.5",
    "@emotion/styled": "11.1.5",
    "react": "17.0.2",

**Comment by vadistic** [comment]
I'm just dealing with the same issue.

It seems the issue is a mismatch between `Interpolation<Theme>` from emotion 11 `@emotion/react` and `InterpolationWithTheme<Theme>` from emotion 10 `@emotion/core` that also provides globals to JSX css prop.

It may be happening because storybook depends on emotion 10 so you have it installed.

My fix is to do path-package and comment-out those few lines (82-100) dealing with globals from `@emotion/core/types/index.d.ts`

**Comment by Vadorequest** [comment]
Yeah, it sounds legit, as I use other libs that are still using Emotion 10. (like StoryBook indeed)

> My fix is to do patch-package and comment-out those few lines

You mean you edit the installed packages within node_modules by hand? That won't work for me because CI/CD builds won't do that and it'll fail during build due to TS error, so my best course of action is still to `@ts-ignore` those.

**Comment by michaeljaltamirano** [comment]
> Yeah, it sounds legit, as I use other libs that are still using Emotion 10. (like StoryBook indeed)
> 
> > My fix is to do patch-package and comment-out those few lines
> 
> You mean you edit the installed packages within node_modules by hand? That won't work for me because CI/CD builds won't do that and it'll fail during build due to TS error, so my best course of action is still to `@ts-ignore` those.

I was able to get around this in a Storybook repo by using `paths` in `tsconfig.json` for the `stories` directory. Maybe this will help you: https://github.com/curology/radiance-ui/pull/550/commits/70caef01a6e550ac2167ddc161bfeb1744f6241d#diff-9f2f153e2e66180524e5c8dff6501791010c150b089485a548c2c92270b00f5eR6-R10

**Comment by Andarist** [comment]
This sounds like an issue with Storybook (since it's still on Emotion 10). Not much that we can do about it here.

**Comment by vadistic** [comment]
> You mean you edit the installed packages within node_modules by hand? That won't work for me because CI/CD builds won't do that and it'll fail during build due to TS error, so my best course of action is still to @ts-ignore those.

Yep, but then you're creating a patch that can be applied in any next install - eg on CI.

This can be done with `patch-package` tool:

https://www.npmjs.com/package/patch-package

Or you can use yarn 2 patch feature (if you're using it): 
https://yarnpkg.com/cli/patch
https://yarnpkg.com/cli/patch-commit
https://yarnpkg.com/features/protocols#patch

**Comment by Vadorequest** [comment]
@Andarist Agreed, nothing to be done from Emotion standpoint besides having a dedicated thread with work arounds and explanations. I guess we should just keep this issue opened until it's solved, it'll help people looking for a workaround until then.

@vadistic Thanks for the workaround, I wasn't aware that kind of patching was possible, great to know!

**Comment by jrolfs** [comment]
My solution was to add `@emotion/core/types/index.d.ts` to `.yarnclean` ... obviously only works if you're using Yarn.

**Comment by dwjohnston** [comment]
Oh wow - it looks like I've possibly stumbled on the right place for me issue. 

Just incase anyone else is searching the same things I am, the css props looks like: 

```
(property) css?: ((string | number | boolean | ComponentSelector | Keyframes | SerializedStyles | ArrayInterpolation<undefined> | ObjectInterpolation<...> | ((theme: any) => Interpolation<...>)) & (string | ... 7 more ... | FunctionInterpolation<...>)) | null | undefined
```

where it should look like: 

```
(property) css?: Interpolation<Theme>
```


@jrolfs  solution semi works for me - if I manually delete the file the types start working again.
